spec/factories/course_forum_topics.rb

```diff
@@ -1,12 +1,20 @@
 # frozen_string_literal: true
 FactoryGirl.define do
   factory :forum_topic, class: Course::Forum::Topic.name do
-    forum
+    transient do
+      course { build(:course) }
```

Comment:
Hmm true, not sure how we can handle the issue where `course_discussion_topic` not being a factory. 

Actually, there is no need for `parent: :course_discussion_topic`, rather you can add it as an additional attribute for the factory (I think): 

```ruby
FactoryGirl.define do
  factory :forum_topic, class: Course::Forum::Topic.name do 
    course_discussion_topic # not sure whether this is an attribute, or transient. 
    ...
  end
end
```